rpc/RPCTransaction.java

```diff
@@ -121,12 +118,6 @@ public void commit() {
                 .putAllMetadata(tracingData())
                 .setCommitReq(TransactionProto.Transaction.Commit.Req.getDefaultInstance());
         execute(commitReq);
-        // TODO: Currently this close request is parsed as an "unknown frame" on the server because the server
-        //  terminates the stream when commit is called. Three possible solutions are:
-        //  1) don't close the transaction, or the stream, on commit
-        //  2) let the server take full responsibility for closing the stream on commit
-        //  3) let the client take full responsibility for closing the stream on commit
-        close();
```

Comment:
This call to `close()` is unneeded as the server closes the transaction itself on a successful commit. Essentially this implements solution (2) from the comments: `let the server take full responsibility for closing the stream on commit`.

The client should be as dumb as possible; this change helps to achieve that.